cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -45,6 +47,26 @@ const (
 	MaxNodeStartupTime = 15 * time.Minute
 )
 
+type maxNodeProvisionTimeProvider interface {
+	// GetMaxNodeProvisionTime returns MaxNodeProvisionTime value that should be used for the given NodeGroup.
+	GetMaxNodeProvisionTime(nodeGroup cloudprovider.NodeGroup) (time.Duration, error)
+}
+
+// NewDefaultMaxNodeProvisionTimeProvider returns the default maxNodeProvisionTimeProvider which uses the NodeGroupConfigProcessor.
+func NewDefaultMaxNodeProvisionTimeProvider(context *context.AutoscalingContext, nodeGroupConfigProcessor nodegroupconfig.NodeGroupConfigProcessor) maxNodeProvisionTimeProvider {
```

Comment:
This doesn't really have much to do with clusterstate, so I'd move it out. Separate package might be an overkill, but perhaps a separate file?